pagopa/io-app
# chore: [PE-803] CGN discount outcome screen

## Short description
This pull request introduces functionality for generating a new discount code when the previous one has expired with error handling

## List of changes proposed in this pull request
- Added new routes for the expired discount code screen in the navigation files
- Updated `CgnDiscountCodeScreen` to include logic for generating new discount codes and handling expired codes 
- Replace legacy `FooterActions` with `io-app-design-system` component
- Reset discount code value at timer expiration 

## How to test
(From dev, for better test experience, ensure `discountTypes` is a single array containing only the value `api`)

- Try to get a discount code with expiration 
- After the timer expires, ensure the appropriate outcome screen is displayed correctly
- Verify if a new discount code is generated pressing on `Usa un nuovo codice`
- Confirm the system navigates to the correct error screen when a discount code cannot be generated pressing `Usa un nuovo codice`

## Preview
https://github.com/user-attachments/assets/6ec8010a-654c-445d-ba1e-782d8a8b8b1b
